system/util/dummy_diag_publisher/src/dummy_diag_publisher_node/dummy_diag_publisher_node.cpp

```diff
@@ -14,47 +14,63 @@
  * limitations under the License.
  */
 
-#include <dummy_diag_publisher/dummy_diag_publisher_node.h>
+#include "dummy_diag_publisher/dummy_diag_publisher_node.h"
 
-#include <boost/bind.hpp>
-
-#include <fmt/format.h>
+#include <rclcpp/create_timer.hpp>
 
 namespace
 {
-template <class Config>
-Config getConfig(const ros::NodeHandle & nh, const std::string & config_name)
+template <typename T>
+void update_param(
+  const std::vector<rclcpp::Parameter> & parameters, const std::string & name, T & value)
 {
-  XmlRpc::XmlRpcValue xml;
-  if (!nh.getParam(config_name, xml)) {
-    throw std::runtime_error(fmt::format("no parameter found: {}", config_name));
+  auto it = std::find_if(
+    parameters.cbegin(), parameters.cend(),
+    [&name](const rclcpp::Parameter & parameter) { return parameter.get_name() == name; });
+  if (it != parameters.cend()) {
+    value = it->template get_value<T>();
   }
-
-  return Config{xml};
 }
 }  // namespace
 
-void DummyDiagPublisherNode::onConfig(
-  const dummy_diag_publisher::DummyDiagPublisherConfig & config, const uint32_t level)
+rcl_interfaces::msg::SetParametersResult DummyDiagPublisherNode::paramCallback(
+  const std::vector<rclcpp::Parameter> & parameters)
 {
-  config_ = config;
+  
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = true;
+  result.reason = "success";
+
+  DummyDiagPublisherConfig config = config_;
+  try {
+    int status;
+    update_param(parameters, "status", status);
```

Comment:
`status` must be initialized with original status because update_param only updates when the parameter is actually set.
e.g. if we only set `is_active` parameter in commandline, `status` will have undefined value after update_param.

 ```suggestion
    int status = static_cast<int>(config.status);
    update_param(parameters, "status", status);
```